public/app/plugins/datasource/loki/querySplitting.ts

```diff
@@ -173,7 +173,8 @@ function getNextRequestPointers(requests: LokiGroupedRequest[], requestGroup: nu
 
 export function runSplitQuery(datasource: LokiDatasource, request: DataQueryRequest<LokiQuery>) {
   const queries = request.targets.filter((query) => !query.hide);
-  const [instantQueries, normalQueries] = partition(queries, (query) => query.queryType === LokiQueryType.Instant);
+  const [instantQueries, nonInstantQueries] = partition(queries, (query) => query.queryType === LokiQueryType.Instant);
+  const [distinctQueries, normalQueries] = partition(nonInstantQueries, (query) => isQueryWithDistinct(query.expr));
```

Comment:
Is there any difference between how do we process and run `instant` queries and queries with `distinct`? I am just wondering, if we couldn't unify this into something like `queriesNotSuitableForSplitting`?  And have `isQuerySuitableForSplitting` function with all of the checks, such as: 
- non-instant query
- not a query with distinct

Reply:
There's no difference on how we treat those, so this is a good idea.

Reply:
Done.